server/etcdmain/config.go

```diff
@@ -260,6 +260,13 @@ func (cfg *config) configFromCmdLine() error {
 		cfg.ec.InitialCluster = ""
 	}
 
+	// make sure there is no conflict in the flag settings in the ExperimentalFlagMigrationMap
+	for oldFlag, newFlag := range embed.ExperimentalFlagMigrationMap {
+		if flags.IsSet(cfg.cf.flagSet, oldFlag) && flags.IsSet(cfg.cf.flagSet, newFlag) {
+			return fmt.Errorf("cannot set --%s and --%s at the same time, please use --%s only", oldFlag, newFlag, newFlag)
+		}
+	}
```

Comment:
Similar to my comment above, suggest to move this to `parse`, in order to take care of both command-line and config file cases.